singularity-eos/eos/eos_polynomial.hpp

```diff
@@ -0,0 +1,212 @@
+//------------------------------------------------------------------------------
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_EOS_POLYNOMIAL_HPP_
+#define _SINGULARITY_EOS_EOS_EOS_POLYNOMIAL_HPP_
+
+// stdlib
+#include <cmath>
+#include <cstdio>
+#include <string>
+
+// Ports-of-call
+#include <ports-of-call/portability.hpp>
+
+// Base stuff
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/base/eos_error.hpp>
+#include <singularity-eos/base/robust_utils.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+
+class Polynomial : public EosBase<Polynomial> {
+ public:
+  Polynomial() = default;
+  PORTABLE_INLINE_FUNCTION Polynomial(Real rho0,
+                                      Real a0, Real a1, Real a2c, Real a2e, Real a3,
+                                      Real b0, Real b1, Real b2c, Real b2e, Real b3)
+    : _rho0(rho0),
+      _a0(a0), _a1(a1), _a2c(a2c), _a2e(a2e), _a3(a3),  
+      _b0(b0), _b1(b1), _b2c(b2c), _b2e(b2e), _b3(b3)  {
+    checkParams();
+  }
+
+  Polynomial GetOnDevice() { return *this; }
+  PORTABLE_INLINE_FUNCTION Real TemperatureFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return 0.0;
+  }
+  PORTABLE_INLINE_FUNCTION void checkParams() const {
+    // Portable_require seems to do the opposite of what it should. Conditions
+    // reflect this and the code should be changed when ports-of-call changes
+    // PORTABLE_ALWAYS_REQUIRE(_Cv >= 0, "Heat capacity must be positive");
+    // PORTABLE_ALWAYS_REQUIRE(_gm1 >= 0, "Gruneisen parameter must be positive");
+    // PORTABLE_ALWAYS_REQUIRE(_EntropyT0 >= 0,
+    //                         "Entropy reference temperature must be positive");
+    // PORTABLE_ALWAYS_REQUIRE(_EntropyRho0 >= 0,
+    //                         "Entropy reference density must be positive");
```

Comment:
I thought we fixed that in `ports_of_call`, but I think `singularity-eos` has not yet updated to the current version. This should happen soon hopefully. @mauneyc-LANL has a WIP PR in.

Reply:
Ignore the comment -- it was copy-pasta from another test